### ISSUE TITLE
V1.5.3 Automatic compression of large response body?

### ISSUE THREAD

**Issue by yangjinheng** [zeromicro/go-zero#3399]
**Describe the bug**

I used httputil.NewSingleHostReverseProxy to proxy access to prometheus, which worked fine in v1.5.2, but on v1.5.3, the large response body was automatically compressed into gzip. What caused this change?

```
The response formats of different versions have changed, just like: 

   v1.5.2
  reponse: {"hello world"}
  
  v1.5.3
  reponse: H4sIAAAAAAAAA6tWykjNyclXKM8vyklRquUCAPjSHAQQAAAA

  $ echo H4sIAAAAAAAAA6tWykjNyclXKM8vyklRquUCAPjSHAQQAAAA | base64 -d | zcat
  {"hello world"}
  
  Must be processed with gzip, but response headers does not have a gzip flag ??
```

**To Reproduce**

Use httputil.NewSingleHostReverseProxy to forward requests to prometheus as handler using rest.Route wrapper

**Environments (please complete the following information):**
 - OS: [e.g. Linux] Ubuntu
 - go-zero version [e.g. 1.2.1] v1.5.3
 - goctl version [e.g. 1.2.1, optional] v1.5.2

**Comment by kevwan** [comment]
Any demo code for this? We didn't enable compression by default.

**Comment by kevwan** [comment]
@yangjinheng any update on this issue?

**Comment by yangjinheng** [comment]
I rolled back the version, too busy to take care of it, It looks like this.

~~~go
func forwardPrometheus(svcCtx *svc.ServiceContext) ([]rest.Route, error) {
	u, err := url.Parse(svcCtx.Config.Prometheus.Addr)
	if err != nil {
		return nil, err
	}

	var spath string
	u.Path, spath = spath, u.Path

	routes := []rest.Route{}
	proxy := httputil.NewSingleHostReverseProxy(u)

	for i := 0; i < 10; i++ {
		routes = append(routes,
			rest.Route{
				Method:  http.MethodGet,
				Path:    spath,
				Handler: proxy.ServeHTTP,
			},
			rest.Route{
				Method:  http.MethodPost,
				Path:    spath,
				Handler: proxy.ServeHTTP,
			},
		)
		spath = spath + "/:"
	}

	return rest.WithMiddlewares([]rest.Middleware{svcCtx.Authorization}, routes...), nil
}
~~~

**Comment by yangjinheng** [comment]
the problem is fixed in v1.5.4
